tests/test_csc.py

```diff
@@ -111,11 +129,40 @@ async def test_stop_propagate_laser(self):
                 topic=self.remote.evt_detailedState,
                 detailedState=TunableLaser.LaserDetailedState.NONPROPAGATING,
             )
+            with self.subTest("Test when not propagating"):
+                with self.assertRaises(salobj.AckError):
+                    await self.remote.cmd_stopPropagateLaser.set_start(
+                        timeout=STD_TIMEOUT
+                    )
+            with self.subTest("Test when disconnecting"):
+                await self.remote.cmd_startPropagateLaser.set_start(timeout=STD_TIMEOUT)
+                await self.remote.cmd_disable.set_start(timeout=STD_TIMEOUT)
+                await self.remote.cmd_standby.set_start(timeout=STD_TIMEOUT)
 
     async def test_clear_laser_fault(self):
         async with self.make_csc(initial_state=salobj.State.ENABLED, simulation_mode=1):
+            self.csc.simulator.device.do_change_m_cpu800_18_power("FAULT")
             await self.remote.cmd_clearLaserFault.set_start(timeout=STD_TIMEOUT)
 
+    async def test_set_burst_mode(self):
+        async with self.make_csc(initial_state=salobj.State.ENABLED, simulation_mode=1):
+            await self.remote.cmd_setBurstMode.set_start(timeout=STD_TIMEOUT)
+            await self.assert_next_sample(topic=self.remote.evt_burstModeSet)
+
+    async def test_set_continuous_mode(self):
+        async with self.make_csc(initial_state=salobj.State.ENABLED, simulation_mode=1):
+            await self.remote.cmd_setContinuousMode.set_start(timeout=STD_TIMEOUT)
+            await self.assert_next_sample(topic=self.remote.evt_continuousModeSet)
+
+    async def test_set_burst_count(self):
+        async with self.make_csc(initial_state=salobj.State.ENABLED, simulation_mode=1):
+            await self.remote.cmd_setBurstCount.set_start(count=60, timeout=STD_TIMEOUT)
+            await self.assert_next_sample(topic=self.remote.evt_burstCountSet, count=60)
+
+    async def test_get_config_pkg(self):
+        async with self.make_csc(initial_state=salobj.State.ENABLED, simulation_mode=1):
+            self.assertEqual("ts_config_mtcalsys", self.csc.get_config_pkg())
```

Comment:
Consider `assert self.csc.get_config_pkg() == "ts_config_mtcalsys"`
as shorter and much easier to read.